service/history/mutableStateBuilder.go

```diff
@@ -1921,6 +1930,18 @@ func (e *mutableStateBuilder) addBinaryCheckSumIfNotExists(
 	exeInfo.AutoResetPoints = &workflow.ResetPoints{
 		Points: currResetPoints,
 	}
+	bytes, err := json.Marshal(recentBinaryChecksums)
+	if err != nil {
+		return err
+	}
+	if exeInfo.SearchAttributes == nil {
```

Comment:
when will the search attributes be nil?

Reply:
in general, we should avoid checking nil every in the code
plz add the initialization in the new function as well as load function